aptos-move/aptos-release-builder/src/simulate.rs

```diff
@@ -0,0 +1,500 @@
+        anyhow!("`fun resolve_multi_step_proposal` must have a Move-defined body")
+    })?;
+    code.code = vec![
+        MoveLoc(1),                     /* Push (signer_address: address) onto the stack */
+        Call(create_signer_handle_idx), /* Call create_signer */
+        Ret,
+    ];
+
+    move_bytecode_verifier::verify_module(&m)?;
+
+    let mut blob = vec![];
+    m.serialize(&mut blob)?;
+
+    state_view.set_state_value(
+        StateKey::module_id(&MODULE_ID_APTOS_GOVERNANCE),
+        StateValue::new_legacy(blob.into()),
+    );
+
+    Ok(())
+}
+
+/***************************************************************************************************
+ * Simulation Workflow
+ *
+ **************************************************************************************************/
+pub async fn simulate_multistep_proposal(remote_url: Url, proposal_path: &Path) -> Result<()> {
+    // List all governance scripts
+    let scripts = list_scripts(proposal_path)?;
+    if scripts.is_empty() {
+        bail!("no scripts found")
+    }
+
+    println!("Found {} scripts", scripts.len());
+    for path in &scripts {
+        println!("    {}", path.display());
+    }
+
+    // Compile all scripts
+    println!("Compiling scripts...");
+    let mut compiled_scripts = vec![];
+    for path in &scripts {
+        let framework_package_args = FrameworkPackageArgs::try_parse_from([
+            "dummy_executable_name",
+            "--framework-local-dir",
+            &aptos_framework_path().to_string_lossy(),
+            "--skip-fetch-latest-git-deps",
+        ])?;
+
+        let (blob, _hash) = compile_in_temp_dir(
+            "script",
+            path,
+            &framework_package_args,
+            PromptOptions::yes(),
+            None,
+        )?;
+
+        compiled_scripts.push(blob);
+    }
+
+    // Set up the simulation state view
+    let client = Client::new(remote_url);
+    let debugger = AptosDebugger::rest_client(client.clone())?;
+    let state = client.get_ledger_information().await?.into_inner();
+
+    let state_view = SimulationStateView {
+        remote: &debugger.state_view_at_version(state.version),
+        states: Mutex::new(HashMap::new()),
+    };
+
+    // Patch framework functions to skip the governance process
+    print!("Patching framework functions to bypass governance.. ");
+    std::io::stdout().flush()?;
+
+    patch_create_signer(&state_view)?;
+    patch_resolve_multistep_proposal(&state_view)?;
+    println!("done");
+
+    // Create and fund a sender account that is used to send the governance scripts
+    print!("Creating and funding sender account.. ");
+    std::io::stdout().flush()?;
+    let mut rng = aptos_keygen::KeyGen::from_seed([0; 32]);
+    let balance = 100 * 1_0000_0000; // 100 APT
+    let account = AccountData::new_from_seed(&mut rng, balance, 0);
+    state_view.apply_write_set(account.to_writeset());
+    // TODO: should update coin info
```

Comment:
What does this mean? Do you mean the total supply?

Reply:
Yes, the total supply.